src/analysis/retail/monk/mistweaver/normalizers/CastLinkNormalizer.ts

```diff
@@ -387,6 +387,22 @@ export function isFromRapidDiffusion(event: ApplyBuffEvent | RefreshBuffEvent) {
   return HasRelatedEvent(event, FROM_RAPID_DIFFUSION);
 }
 
+export function isFromRapidDiffusionRisingSunKick(event: ApplyBuffEvent | RefreshBuffEvent) {
+  const rdSourceEvent = GetRelatedEvents(event, FROM_RAPID_DIFFUSION);
+  return (
+    rdSourceEvent[0].type === EventType.Cast &&
+    rdSourceEvent[0].ability.guid === TALENTS_MONK.RISING_SUN_KICK_TALENT.id
+  );
+}
+
+export function isFromRapidDiffusionEnvelopingMist(event: ApplyBuffEvent | RefreshBuffEvent) {
+  const rdSourceEvent = GetRelatedEvents(event, FROM_RAPID_DIFFUSION);
+  return (
+    rdSourceEvent[0].type === EventType.Cast &&
+    rdSourceEvent[0].ability.guid === TALENTS_MONK.ENVELOPING_MIST_TALENT.id
+  );
```

Comment:
In theory, this could throw an error if theres some linking mishaps or called incorrectly. Might want to add some safety handling if rdSourceEvent is empty